File: app/src/main/java/ru/gsench/githubusers/domain/SystemInterface.java
package ru.gsench.githubusers.domain;

import java.io.IOException;
import java.net.URL;

import ru.gsench.githubusers.domain.utils.HttpParams;
import ru.gsench.githubusers.domain.utils.Pair;
import ru.gsench.githubusers.domain.utils.function;

/**
 * Created by grish on 01.05.2017.
 */

public interface SystemInterface {

    public void doOnBackground(function<Void> background);
    public void doOnForeground(function<Void> function);
    public Pair<byte[], HttpParams> httpGet(URL url, HttpParams params) throws IOException;
    public String[] getSavedStringArray(String title, String[] def);
    public void saveStringArray(String title, String[] array);
    /**
    public String getSavedString(String title, String def);
    public void saveString(String title, String string);
    public int getSavedInt(String title, int def);
    public void saveInt(String title, int i);*/
    public void removeSaved(String str);

}

File: app/src/main/java/ru/gsench/githubusers/domain/SystemInterface.kt
package ru.gsench.githubusers.domain

import java.io.IOException
import java.net.URL

import ru.gsench.githubusers.domain.utils.HttpParams
import ru.gsench.githubusers.domain.utils.Pair
import ru.gsench.githubusers.domain.utils.function

/**
 * Created by grish on 01.05.2017.
 */

interface SystemInterface {

    fun doOnBackground(background: ()->Unit)
    fun doOnForeground(foreground: ()->Unit)
    @Throws(IOException::class)
    fun httpGet(url: URL, params: HttpParams?): Pair<ByteArray, HttpParams>

    fun getSavedStringArray(title: String, def: Array<String>?): Array<String>
    fun saveStringArray(title: String, array: Array<String>)
    /**
     * public String getSavedString(String title, String def);
     * public void saveString(String title, String string);
     * public int getSavedInt(String title, int def);
     * public void saveInt(String title, int i); */
    fun removeSaved(str: String)

}
